music21/analysis/harmonicFunction.py

```diff
@@ -0,0 +1,305 @@
+    though stictly speaking, it's handled through a special HarmonicFunction enum object.
+
+    >>> fn = analysis.harmonicFunction.HarmonicFunction.TONIC_MAJOR
+    >>> str(fn)
+    'T'
+
+    >>> analysis.harmonicFunction.functionToRoman(fn).figure
+    'I'
+
+    As with Roman numerals, this is case sensitive.
+    For instance, 't' indicates a minor tonic
+    as distinct from the major tonic, 'T'.
+
+    >>> analysis.harmonicFunction.functionToRoman('t').figure
+    'i'
+
+    There are 18 main functional labels supported in all, for
+    the three functional categories
+    (T for tonic, S for subdominant, and D for dominant) and
+    three relevant transformation types (none, P, and G)
+    all in upper and lower case (for major/minor):
+    T, Tp, Tg, t, tP, tG,
+    S, Sp, Sg, s, sP, sG,
+    D, Dp, Dg, d, dP, dG.
+
+    Note that this module uses terminology from modern German music theory
+    where Functional notation ('HarmonicFunctionstheorie') is typically used
+    throughout the curriculum in preference over Roman numerals ('Stufentheorie').
+
+    First, note the false friend: here 'P' for 'Parallel'
+    connects a major triad with the minor triad a minor third below (e.g. C-a).
+    (in English-speaking traditions this would usually be 'relative').
+
+    Second, note that this module uses
+    'G' (and 'g'), standing for
+    'Gegenklänge' or 'Gegenparallelen'.
+    'L' (and 'l') for Leittonwechselklänge is equivalent to this.
+    (Again, 'G' is more common in modern German-language music theory).
+
+    Use the keyOrScale argement to specify a key.
+    This makes a difference where 6th and 7th degrees of minor are involved.
+
+    >>> analysis.harmonicFunction.functionToRoman('sP', keyOrScale='C').figure
+    'bVI'
+
+    >>> analysis.harmonicFunction.functionToRoman('sP', keyOrScale='a').figure
+    'VI'
+
+    Some of the 18 main functions overlap, with two functional labels
+    referring to the same Roman numeral figure.
+    For instance both 'Tg' and 'Dp' simply map to 'iii':
+
+    >>> analysis.harmonicFunction.functionToRoman('Tp').figure
+    'vi'
+
+    >>> analysis.harmonicFunction.functionToRoman('Sg').figure
+    'vi'
+
+    The reverse operation is handled by the complementary
+    :func:`~music21.analysis.harmonicFunction.romanToFunction`.
+    In this case, :func:`~music21.analysis.harmonicFunction.romanToFunction`
+    follows the convention of preferring the P-version over alternatives.
+
+    >>> rn = roman.RomanNumeral('vi')
+    >>> str(analysis.harmonicFunction.romanToFunction(rn))
+    'Tp'
+
+    '''
+    if isinstance(keyOrScale, str):
+        keyOrScale = key.Key(keyOrScale)
+
+    referenceTuples = functionFigureTuplesMajor
+    if keyOrScale.mode == 'minor':
+        referenceTuples = functionFigureTuplesMinor
+
+    for thisKey, thisValue in referenceTuples.items():
+        if thisHarmonicFunction == thisKey:
+            return roman.RomanNumeral(thisValue, keyOrScale)
+
+    return False
```

Comment:
A small performance improvement to avoid walking every item of `referenceTuples`. You could do something similar below. And return `None` instead of mixing booleans with romans.
```suggestion
try:
    figure = referenceTuples[thisHarmonicFunction]
except KeyError:
    return None

return roman.RomanNumeral(figure, keyOrScale)
```